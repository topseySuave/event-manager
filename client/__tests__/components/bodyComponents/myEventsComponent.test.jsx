import React from 'react';
import { shallow } from 'enzyme';
import { MyEventCardHolder } from
  '../../../src/components/bodyComponents/MyEventCardHolder';
import {
  fetchEventRequest,
  loadMoreEvents,
  fetchSessionEventRequest
} from './../../../src/actions/events-actions';

let wrapper, instance, mounted;
let store, comProps;

const getComponent = (props) => {
  mounted = shallow(<MyEventCardHolder {...props} />);
  return mounted;
};

let componentDidMountSpy = jest
  .spyOn(MyEventCardHolder.prototype, 'componentDidMount');
let loadMoreSpy = jest
  .spyOn(MyEventCardHolder.prototype, 'loadMore');

describe('MyEventCardHolder component', () => {
  test('MyEventCardHolder component should mount', () => {
    comProps = {
      fetchSessionEventRequest,
      activeUser: {
        isAuthenticated: true,
        user: { id: 1 }
      }
    };
    wrapper = getComponent(comProps);
    expect(wrapper).toMatchSnapshot();
    expect(componentDidMountSpy).toBeCalled();
  });

  test('component should update props', () => {
    wrapper = getComponent(comProps);
    let newProps = {
      allEvents: {
        sessEvents: {
          meta: {
            page: 1,
            pageCount: 2,
            pageSize: 5,
            totalCount: 5,
            loadingmore: false,
            loadmore: true
          },
          events: [{ id: 1, title: 'one event' }, { id: 2, title: 'two event' }]
        }
      }
    };
    wrapper.setProps(newProps);
    expect(wrapper.state('isLoading')).toBeFalsy();
  });

  test('should set isLoading to false if there are no events', () => {
    wrapper = getComponent(comProps);
    let newProps = {
      allEvents: {
        sessEvents: {
          events: []
        }
      }
    };
    wrapper.setProps(newProps);
    expect(wrapper.state('isLoading')).toBeFalsy();
  });

  test('should be to load more', () => {
    wrapper = getComponent(comProps);
    let newProps = {
      loadMoreEvents,
      allEvents: {
        sessEvents: {
          events: []
        }
      }
    };
    wrapper.setProps(newProps);
    wrapper.instance().loadMore();
    expect(loadMoreSpy).toBeCalled();
  });
});
