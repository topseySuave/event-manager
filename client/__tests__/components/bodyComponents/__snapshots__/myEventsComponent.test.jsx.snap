// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyEventCardHolder component MyEventCardHolder component should mount 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyEventCardHolder
    activeUser={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "id": 1,
        },
      }
    }
    fetchSessionEventRequest={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="container popular__events"
        style={
          Object {
            "minHeight": "90vh",
            "paddingTop": "64px",
          }
        }
      >
        <h4
          className="center-align"
        >
          My Events
        </h4>
        <div
          style={
            Object {
              "height": "500px",
              "marginTop": "100px",
            }
          }
        >
          <CircularLoader />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="center-align"
          >
            My Events
          </h4>,
          <div
            style={
              Object {
                "height": "500px",
                "marginTop": "100px",
              }
            }
          >
            <CircularLoader />
          </div>,
        ],
        "className": "container popular__events",
        "style": Object {
          "minHeight": "90vh",
          "paddingTop": "64px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Events",
            "className": "center-align",
          },
          "ref": null,
          "rendered": "My Events",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CircularLoader />,
            "style": Object {
              "height": "500px",
              "marginTop": "100px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="container popular__events"
          style={
            Object {
              "minHeight": "90vh",
              "paddingTop": "64px",
            }
          }
        >
          <h4
            className="center-align"
          >
            My Events
          </h4>
          <div
            style={
              Object {
                "height": "500px",
                "marginTop": "100px",
              }
            }
          >
            <CircularLoader />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="center-align"
            >
              My Events
            </h4>,
            <div
              style={
                Object {
                  "height": "500px",
                  "marginTop": "100px",
                }
              }
            >
              <CircularLoader />
            </div>,
          ],
          "className": "container popular__events",
          "style": Object {
            "minHeight": "90vh",
            "paddingTop": "64px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "My Events",
              "className": "center-align",
            },
            "ref": null,
            "rendered": "My Events",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <CircularLoader />,
              "style": Object {
                "height": "500px",
                "marginTop": "100px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
